servicetalk-http-api/src/main/java/io/servicetalk/http/api/SpecialHttpExecutionStrategy.java

```diff
@@ -112,7 +112,7 @@ public boolean isCloseOffloaded() {
          */
         @Override
         public HttpExecutionStrategy merge(final HttpExecutionStrategy other) {
-            return other;
+            throw new UnsupportedOperationException("must not be merged");
```

Comment:
Can you please elaborate what is the benefit of throwing an exception from here? Because we change the behavior, it introduces a low risk of blowing up at runtime if users used `defaultStrategy().merge(...)` somewhere. Instead of throwing here, consider logging at info/warn level and keeping the pre-existing behavior. Understand that it may lead to unexpected results, but it will still work instead of failing. Offloading is just an optimization.

If you would like to verify that ST never invokes this method in tests, consider using `assert`.

Reply:
I will switch to `assert` for now.